conversation_reconstruction_dataflow/conversation_reconstruction/dataflow_main.py

```diff
@@ -11,81 +11,100 @@
-                   | beam.io.Write(bigquery.BigQuerySink(known_args.output_table, schema=known_args.schema, validate=True)))
+                   | beam.io.Write(bigquery_io.BigQuerySink(known_args.output_table, schema=known_args.output_schema, validate=True)))
 
 class ReconstructConversation(beam.DoFn):
-  def process(self, element):
+  def process(self, row):
 
-    from construct_utils import constructing_pipeline
+#    from construct_utils import constructing_pipeline
     import logging
-    import json
+    from google.cloud import bigquery as bigquery_op 
+    input_table = "wikidetox-viz:wikidetox_conversations.test_page_3_issue21" 
 
     logging.info('USERLOG: Work start')
+    page_id = row['page_id']
+    client = bigquery_op.Client(project='wikidetox-viz')
+    query = ("SELECT rev_id FROM %s WHERE page_id = \"%s\""%(input_table, page_id))
+    query_job = client.query(query)
+    rev_ids = []
+    for row in query_job:
+        rev_ids.append(row['rev_id'])
 
     return 
-    page = json.loads(element)
-    logging.info('USERLOG: Working on %s' % page['page_id'])
-    page_id = page['page_id']
-    status = 'NO STATUS'
-
-    local_out_filename = page_id + '.json'
-    out_file_path = 'gs://wikidetox-viz-dataflow/conversations/'
-
-    check_file_cmd = (['gsutil', '-q', 'stat', path.join(out_file_path, local_out_filename)])
-    file_not_exist = subprocess.call(check_file_cmd)
-    if '/Archive' in page['page_title']:
-       logging.info('USERLOG: SKIPPED FILE %s as it is an archived talk page.' % page_id)
-       status = 'ARCHIVE PAGE'
-    if(file_not_exist and not(status == 'ARCHIVE PAGE')):
-      try:
-        logging.info('USERLOG: Loading constructor with input: %s output: %s' % (page_id, local_out_filename))
-        processor = constructing_pipeline.ConstructingPipeline(page, local_out_filename)
-        logging.info('USERLOG: Running constructor on %s.' % page_id)
-        processor.run_constructor()
-
-        logging.info('USERLOG: Running gsutil cp %s %s' % (local_out_filename, out_file_path))
-        cp_remote_cmd = (['gsutil', 'cp', local_out_filename, out_file_path])
-        cp_proc = subprocess.call(cp_remote_cmd)
-        if cp_proc == 0:
-          status = 'SUCCESS'
-        else:
-          status = 'FAILED to copy to remote'
-
-        logging.info('USERLOG: Removing local files.')
-        rm_cmd = (['rm', local_out_filename])
-        subprocess.call(rm_cmd)
-        rm_cmd = (['rm', chunk_name])
-        subprocess.call(rm_cmd)
-
-        logging.info('USERLOG: Job complete on %s.' % page_id)
-
-      except:
-        logging.info('USERLOG: Hit unknown exception on %s.' % page_id)
-        status = 'FAILED with Unknown Exception'
-    else:
-      logging.info('USERLOG: SKIPPED FILE %s as it is already processed.' % page_id)
-      status = 'ALREADY EXISTS'
-
-    return 
+    construction_cmd = ['python2', '-m', 'construct_utils.run_constructor', '--table', input_table, '--revisions', rev_ids]
```

Comment:
Can I use python3 here? Some packages I used in the reconstruction process can not be run under python2.